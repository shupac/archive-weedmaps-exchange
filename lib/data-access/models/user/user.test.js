import { getSnapshot } from 'mobx-state-tree';
import User from './index';

const mockUser = {
  username: 'weedmaps',
  email: 'weedmaps@weedmaps.com',
  roles: ['ops_manager', 'content_moderator', 'sales_manager'],
  slug: 'kushagram',
};
export default mockUser;

describe('User Model', () => {
  it('can create an instance of a model', () => {
    const user = User.create(mockUser);
    expect(getSnapshot(user)).toMatchSnapshot();
  });

  it('can get the proper roles of a user', () => {
    const user = User.create(mockUser);
    expect(user.isOpsManager).toEqual(true);
    expect(user.isSalesManager).toEqual(true);
    expect(user.isModerator).toEqual(true);
    expect(user.isListingOwner).toEqual(false);
  });
});
