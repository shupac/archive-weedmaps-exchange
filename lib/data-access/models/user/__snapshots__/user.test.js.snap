// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Model can create an instance of a model 1`] = `
Object {
  "email": "weedmaps@weedmaps.com",
  "roles": Array [
    "ops_manager",
    "content_moderator",
    "sales_manager",
  ],
  "slug": "kushagram",
  "username": "weedmaps",
}
`;
